### PR TITLE
Fix alignment offline validation ROOT6.27/01 issue

### PR DESCRIPTION
**Pull request by TomasKello**
#### PR description:

Recasting pointer access that was in conflict with `ROOT Version: 6.27/01` and caused unit test failure.

#### PR validation:

Unit tests now successful for `el8_amd64_gcc10` and both `ROOT Version: 6.27/01` and `ROOT Version: 6.24/07`.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test for CMSSW_12_6_ROOT6_X

File: Alignment/OfflineValidation/src/PlotAlignmentValidation.cc [hunk 1]
```diff
@@ -1229,24 +1229,19 @@ void PlotAlignmentValidation::plotChi2(const char* inputFile) {
...
-        }
-      }
+    if (fi1->GetDirectory("TrackerOfflineValidation/GlobalTrackVariables") == nullptr) {
+      errorflag = kTRUE;
     }
+  } else {
+    errorflag = kTRUE;
+  }
+  TCanvas* normchi = new TCanvas((TCanvas*)fi1->Get("TrackerOfflineValidation/GlobalTrackVariables/h_normchi2"));
+  TCanvas* chiprob = new TCanvas((TCanvas*)fi1->Get("TrackerOfflineValidation/GlobalTrackVariables/h_chi2Prob"));
```
Comment (mmusich) [inline Alignment/OfflineValidation/src/PlotAlignmentValidation.cc:1242]: why do you need `new` here? Can't you just assign after cast?

**Comment by tvami** [comment]
" Unable to find CMSSW IB for CMSSW_12_6_ROOT6_X" ? that doesnt seem right, @smuzaffar ?

**Comment by tvami** [comment]
there is a `CMSSW_12_6_ROOT6_X_2022-11-17-2300`

File: Alignment/OfflineValidation/src/PlotAlignmentValidation.cc (hunk 1 above)
Comment (pcanal) [inline Alignment/OfflineValidation/src/PlotAlignmentValidation.cc:1242]:
```suggestion
  auto normchi = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_normchi2"));
  auto chiprob = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_chi2Prob"));
```

Comment (TomasKello) [reply Alignment/OfflineValidation/src/PlotAlignmentValidation.cc:1242]: Yes indeed, my initial solution was using wrong constructor. Thanks for spotting. I need to run in through the all tests again and will commit once more.

Comment (pcanal) [reply Alignment/OfflineValidation/src/PlotAlignmentValidation.cc:1242]:
```suggestion
  auto normchi = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_normchi2"));
  normchi->GetFrame()->ResetBit(kCanDelete);
  auto chiprob = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_chi2Prob"));
  chiprob->GetFrame()->ResetBit(kCanDelete);
```
is likely to be fixing the issue.

Comment (TomasKello) [reply Alignment/OfflineValidation/src/PlotAlignmentValidation.cc:1242]: I take back, still crashing.

Comment (TomasKello) [reply Alignment/OfflineValidation/src/PlotAlignmentValidation.cc:1242]: Ah great, I see your comment on bug in TPad https://github.com/cms-sw/cmssw/issues/40091#issuecomment-1320661264
It works with a workaround solution. I will make a comment to remove this extra step in this code with future ROOT version.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
> Unable to find CMSSW IB for CMSSW_12_6_ROOT6_X" ? that doesnt seem right

How can we test in a realistic setup ?

**Comment by tvami** [comment]
+alca
 * no dedicated IBs.. but the default IB tests do pass

**Comment by perrotta** [comment]
+1

**Pull request merged**